### PR TITLE
Makes gutlunches hungry for gibs again

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
In my zealousness in #31141 I mistakenly removed the ability of gutlunches to find and eat gibs. As gib-eating directly affects their milk production, this is a bit worse than just some missing fluff.

[Changelogs]: 

:cl: Naksu
fix: Gutlunches will now once again look for gibs to eat.
/:cl:

[why]:
Bugfix

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm [hunk 1]
```diff
@@ -62,6 +62,10 @@
 
 		return TRUE
 
+	if(isobj(the_target))
+		if(attack_all_objects || is_type_in_typecache(the_target, wanted_objects))
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm]: Combine these ifs

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm (hunk 1 above)
Comment (vuonojenmustaturska) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm]: I'll do one better

**Comment by FantasticFwoosh** [comment]
Was there any other collateral damage in the behaviour of gold-grubs etc? Its one of the only other object orientated creatures that comes to mind.

**Comment by JJRcop** [comment]
optional name here does it again!

**Comment by vuonojenmustaturska** [comment]
@FantasticFwoosh no, goldgrubs are straight simple animals AI-wise. The mistake with gutlunches was when I ported their snowflake exceptions to a copy-pasted override proc and trimmed down on all the stuff gutlunches don't actually do.

**Pull request merged**